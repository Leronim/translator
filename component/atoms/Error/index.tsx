import React from 'react';
import { View, Text, StyleSheet } from 'react-native';

interface Props {
    message: string;
    color: string;
}

const Error: React.FC<Props> = ({ message, color }) => {
    const style = StyleSheet.create({
        text: {
            color: color,
            fontWeight: "400",
            fontSize: 24
        },
        container: {
            justifyContent: 'center',
            flex: 1,
            alignItems: 'center'
        }
    })


    return (
        <View style={style.container}>
            <Text style={style.text} testID="errorMessage">{message}</Text>
        </View>
    )
}

export default Error;