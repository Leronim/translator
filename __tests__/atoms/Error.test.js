import React from 'react';
import Error from '../../component/atoms/Error';
import { render } from '@testing-library/react-native';

test('Erorr render', () => {
    const tree = render(<Error message="Something wrong!" color="#FF0000"/>).toJSON();
    expect(tree).toMatchSnapshot();
})

test('Error props', () => {
    const message = "Something wrong!";
    const color = "#FF0000";

    const { getByTestId } = render(<Error message={message} color={color}/>);

    expect(getByTestId('errorMessage').children[0]).toEqual(message);
    expect(getByTestId('errorMessage').props.style.color).toEqual(color)
})