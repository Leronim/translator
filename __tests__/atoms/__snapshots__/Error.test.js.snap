// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Erorr render 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      Object {
        "color": "#FF0000",
        "fontSize": 24,
        "fontWeight": "400",
      }
    }
    testID="errorMessage"
  >
    Something wrong!
  </Text>
</View>
`;
